aion/lib/aion/room_channel/monitor.ex

```diff
@@ -57,6 +61,15 @@ defmodule Aion.RoomChannel.Monitor do
   # Game specific interface #
   ###########################
 
+  def get_player_counts do
+    Aion.RoomChannel.Supervisor
+    |> Supervisor.which_children
+    |> Enum.map(fn({_id, pid, _type, _modules}) -> pid end)
+    |> Enum.map(fn(pid) -> {pid, get_scores(pid)} end)
+    |> Enum.map(fn({pid, scores}) -> {get_room_id(pid), Enum.count(scores)} end)
+    |> Map.new
```

Comment:
Why not
```ex
|> Enum.map(fn({_id, pid, _type, _modules}) -> {get_room_id(pid), get_scores(pid) |> Enum.count} end)
```
Or at least
```ex
|> Enum.map(fn({_id, pid, _type, _modules}) -> {get_room_id(pid), get_scores(pid)} end)
|> Enum.map(fn({_room_id, scores} -> {_room_id, Enum.count(scores)}
```

I feel like current solution has redundant transitions (first we return only pid, then we return tuple with pid, and then we finally modify pid instead of just return tuple with room_id in first transition which seems weird).

Reply:
I wanted it to stay legible and show what changes are done one after another, but it's true that all of that could be reduced. I will reduce it to two clauses then. If anything, I would object against using `Enum.count` here due to performance reasons.